#### Mizzare Corporation has collected data on customer satisfaction levels for two types of smart home devices: Smart Thermostats and Smart Lights. They want to determine if there's a significant association between the type of device purchased and the customer's satisfaction level.
#### Objective: To use the Chi-Square test for independence to determine if there's a significant association between the type of smart home device purchased (Smart Thermostats vs. Smart Lights) and the customer satisfaction level.


In [1]:
import numpy as np
import pandas as pd
import scipy.stats as stats

#### 1. First we have to state the NULL hypothesis (H0) and ALTERNATIVE hypothesis (Ha)

In [2]:
# H0 -- There is not a significant association between the type of smart home device purchased (Smart Thermostats vs. Smart Lights) and the customer satisfaction level.
# Ha -- There is a significant association between the type of smart home device purchased (Smart Thermostats vs. Smart Lights) and the customer satisfaction level.
# Given data
n=5
dof=4
# alpha = 5% , CI = 95% 

In [3]:
data = np.array([
    [50, 70],
    [80, 100],
    [60, 90],
    [30, 50],
    [20, 50]
])
satisfaction = ['Very Satisfied', 'Satisfied', 'Neutral', 'Unsatisfied', 'Very Unsatisfied']
devices = ['Smart Thermostat', 'Smart Light']
df = pd.DataFrame(data, index=satisfaction, columns=devices)
df

,Smart Thermostat,Smart Light
Very Satisfied,50,70
Satisfied,80,100
Neutral,60,90
Unsatisfied,30,50
Very Unsatisfied,20,50


#### 2. Now we need to compute the Chi-Square Statistics

In [4]:
chi2_stats=stats.chi2_contingency(df)
chi2_stats

Chi2ContingencyResult(statistic=np.float64(5.638227513227513), pvalue=np.float64(0.22784371130697179), dof=4, expected_freq=array([[ 48.,  72.],
       [ 72., 108.],
       [ 60.,  90.],
       [ 32.,  48.],
       [ 28.,  42.]]))

#### 3.Now determine the Critical Value

In [5]:
chi_critical=stats.chi2.ppf(0.95,4)
chi_critical

np.float64(9.487729036781154)

#### 4.Now we have to make a decision

- Chi-square statistic = 5.64
- Critical value = 9.49
- Since chi_critical > chi_stats , fails to reject the null hypothesis (H0).
- Hence, H0 seems true.There is not a significant association between the type of smart home device purchased (Smart Thermostats vs. Smart Lights) and the customer satisfaction level.